runconfig/parse_unix.go

```diff
@@ -7,51 +7,53 @@ import (
 	"strings"
 )
 
-func parseNetMode(netMode string) (NetworkMode, error) {
-	parts := strings.Split(netMode, ":")
+// ValidateNetMode ensures that the various combinations of requested
+// network settings are valid.
+func ValidateNetMode(c *Config, hc *HostConfig) error {
+	// We may not be passed a host config, such as in the case of docker commit
+	if hc == nil {
+		return nil
+	}
```

Comment:
It is not a CLI backward compatibility issue (as the CLI always defaults to `default`). But it will be an API issue if we dont address it appropriately.

My suggestion is quite simple. You can move this fix : https://github.com/docker/docker/blob/master/daemon/container_unix.go#L918 to `DecodeContainerConfig` just before calling validateNetMode and everything should just work fine without the need to modify the integration-cli tests.